src/context/CategoryContext.js

```diff
@@ -0,0 +1,28 @@
+import axios from "axios";
+import React from "react";
+import { createContext, useContext, useEffect, useState } from "react";
+
+const CategoryContext = createContext();
+const useCategoryContext = () => useContext(CategoryContext);
+
+const CategoryContextProvider = ({ children }) => {
+  const [categoryList, setCategoryList] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        const data = await axios.get("/api/categories");
+        console.log(data.data.categories);
+        setCategoryList(data.data.categories);
+      } catch (err) {
```

Comment:
you can destructure the data here like {data}

Reply:
I don't get it and even didn't know about that. Will try to learn and implement.

Reply:
You can destructure your data like: 

```const { data } = await axios.get("/api/categories")``` 
and then you can remove ```data.data.categories``` to ```data.categories```